api/appendix_h.asciidoc

```diff
@@ -89,7 +89,6 @@ Note that a device that provides the same level of capabilities as an OpenCL 2.x
 | May return:
 
 {CL_DEVICE_ATOMIC_ORDER_RELAXED} \| +
```

Comment:
We have a couple of options for this `CL_DEVICE_ATOMIC_FENCE_CAPABILITIES` query:

1. We could remove it completely, if it's redundant with `CL_DEVICE_ATOMIC_MEMORY_CAPABILITIES`.
2. We could keep it as defined in this PR and remove the requirement to support `CL_DEVICE_ATOMIC_ORDER_ACQ_REL`.
3. We could revert this change, and continue to require `CL_DEVICE_ATOMIC_ORDER_ACQ_REL`, since this is the behavior of the OpenCL 1.2 `mem_fence` built-ins even if they don't use the acquire and release memory orders explicitly.

Note that we have a separate query for the OpenCL C features.  We should also define (or at least understand) the relationship between these atomic queries and the `__opencl_c_atomic_order_acq_rel` feature macro.

Reply:
Thinking on option (1), it seems the cleanest.  The main case that it doesn't support is hardware with strict fences but no way to support strict memory operations (or vice-versa, but that would be more surprising).  I.e. the existing OpenCL 1.2 `mem_fence` problem, or a stronger version of it.

The OpenCL C 3.0 already uses this language for the relevant feature macros (emphasis mine):

> Indicating atomic operations **and** fences support" 

Theoretically other languages, e.g. IL, may not pair these.